File: sdk/dart/test/create_segment_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for CreateSegmentRequest
void main() {
  final instance = CreateSegmentRequest((b) => b
    ..description = 'Segment desc'
    ..rolloutPercent = 50);

  group(CreateSegmentRequest, () {
    test('to test the property `description`', () async {
      expect(instance.description, equals('Segment desc'));
    });

    test('to test the property `rolloutPercent`', () async {
      expect(instance.rolloutPercent, equals(50));
    });
  });
}

File: sdk/java/src/test/java/com/flagent/client/model/CreateSegmentRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for CreateSegmentRequest
 */
class CreateSegmentRequestTest {

    @Test
    void testCreateSegmentRequest() {
        CreateSegmentRequest model = new CreateSegmentRequest()
            .description("New segment")
            .rolloutPercent(100L);
        Assertions.assertNotNull(model);
        Assertions.assertEquals("New segment", model.getDescription());
        Assertions.assertEquals(100L, model.getRolloutPercent());
    }

    @Test
    void descriptionTest() {
        CreateSegmentRequest model = new CreateSegmentRequest().description("Desc");
        Assertions.assertEquals("Desc", model.getDescription());
    }

    @Test
    void rolloutPercentTest() {
        CreateSegmentRequest model = new CreateSegmentRequest().rolloutPercent(50L);
        Assertions.assertEquals(50L, model.getRolloutPercent());
    }

}
